src/v/archival/ntp_archiver_service.cc

```diff
@@ -1813,6 +1817,61 @@ ss::future<> ntp_archiver::housekeeping() {
     }
 }
 
+ss::future<> ntp_archiver::apply_spillover() {
+    const auto manifest_size_limit
+      = config::shard_local_cfg().cloud_storage_spillover_manifest_size.value();
+    if (manifest_size_limit.has_value() == false) {
+        co_return;
+    }
+
+    if (!may_begin_uploads()) {
+        co_return;
+    }
+
+    const auto manifest_upload_timeout
+      = config::shard_local_cfg()
+          .cloud_storage_manifest_upload_timeout_ms.value();
+    const auto manifest_upload_backoff
+      = config::shard_local_cfg().cloud_storage_initial_backoff_ms.value();
+    auto msize = manifest().segments_metadata_bytes();
+    if (msize > manifest_size_limit.value() * 2) {
+        auto tail = [&]() {
+            cloud_storage::spillover_manifest tail(_ntp, _rev);
+            for (const auto& meta : manifest()) {
+                tail.add(meta);
+                if (tail.segments_metadata_bytes() >= manifest_size_limit) {
+                    break;
+                }
+            }
+            return tail;
+        }();
+        vlog(
+          _rtclog.info,
+          "Preparing spillover: manifest has {} segments, spillover "
+          "manifest size: {}, base: {}, last: {}",
+          manifest().size(),
+          tail.size(),
+          tail.get_last_offset(),
+          tail.get_start_offset().value_or(model::offset{}));
+        retry_chain_node upload_rtc(
+          manifest_upload_timeout, manifest_upload_backoff, &_rtcnode);
+        auto res = co_await _remote.upload_manifest(
+          get_bucket_name(), tail, upload_rtc);
+        if (res != cloud_storage::upload_result::success) {
+            vlog(_rtclog.error, "Failed to upload spillover manifest {}", res);
+            co_return;
+        }
```

Comment:
If we crash at this point, we leave an archive manifest in the object store.  When we run again, we _might_ upload the same thing again and proceed normally, but there are other possibilities:
- We might choose a different spillover size next time (e.g. config change) and upload something with the same base offset but different other parameters
- Retention might happen, and we might upload something with a different base offset.

We need to define+test how a reader should interpret the situation where it sees multiple archive manifests that have overlapping offset ranges.

Reply:
This is handled in the follow up PR that implements loading of spillover manifests